azure-iot-hub/azure/iot/hub/iothub_configuration_manager.py

```diff
@@ -0,0 +1,138 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from .auth import ConnectionStringAuthentication
+from .protocol.iot_hub_gateway_service_ap_is20190630 import (
+    IotHubGatewayServiceAPIs20190630 as protocol_client,
+)
+from .protocol.models import (
+    Configuration,
+    ConfigurationContent,
+    ConfigurationQueriesTestInput
+)
+
+
+class IoTHubConfigurationManager(object):
+    """A class to provide convenience APIs for IoTHub Registry Manager operations,
+    based on top of the auto generated IotHub REST APIs
+    """
+
+    def __init__(self, connection_string):
+        """Initializer for a Registry Manager Service client.
+
+        After a successful creation the class has been authenticated with IoTHub and
+        it is ready to call the member APIs to communicate with IoTHub.
+
+        :param str connection_string: The IoTHub connection string used to authenticate connection
+            with IoTHub.
+
+        :returns: Instance of the IoTHubRegistryManager object.
+        :rtype: :class:`azure.iot.hub.IoTHubRegistryManager`
+        """
+
+        self.auth = ConnectionStringAuthentication(connection_string)
+        self.protocol = protocol_client(self.auth, "https://" + self.auth["HostName"]
+        )
+
+    def get_configuration(self, configuration_id):
+        """Retrieves the IoTHub configuration for a particular device.
+
+        :param str configuration_id: The id of the configuration.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
```

Comment:
>`HttpOperationError<msrest.exceptions.HttpOperationError>` [](start = 17, length = 58)

Something to consider is if we want to expose these lower level exceptions as part of our API.

It wouldn't be incorrect to allow this exception from `msrest` to propagate through our API, plenty of Python libraries do such things.

However, we are then making a contract with the user, that this won't change, which is somewhat out of our control. Not sure how strict we need to be about potential breaking changes on service clients though.

Reply:
As we discussed we should rely on the error of MSRest. It is very unlikely they will change this and in service  client we should follow the generated code.

---
In reply to: [354965843](https://github.com/Azure/azure-iot-sdk-python/pull/397#discussion_r354965843) [](ancestors = 354965843)